packages/amplify-graphql-validate-transformer/src/__tests__/minmax-length.test.ts

```diff
@@ -0,0 +1,188 @@
+          });
+        }).toThrow(error);
+      });
+    });
+
+    describe('Negative length values', () => {
+      test.each([
+        {
+          name: 'rejects negative value of "-999999999999999999999999999999"',
+          schema: /* GraphQL */ `
+            type Post @model {
+              id: ID!
+              title: String! @validate(type: minLength, value: "-999999999999999999999999999999")
+            }
+          `,
+          error: "minLength value must be a non-negative integer. Received '-999999999999999999999999999999' for field 'title'",
+        },
+        {
+          name: 'rejects negative value of "-10"',
+          schema: /* GraphQL */ `
+            type Post @model {
+              id: ID!
+              title: String! @validate(type: maxLength, value: "-10")
+            }
+          `,
+          error: "maxLength value must be a non-negative integer. Received '-10' for field 'title'",
+        },
+      ])('$name', ({ schema, error }) => {
+        const transformer = new ValidateTransformer();
+        expect(() => {
+          testTransform({
+            schema,
+            transformers: [new ModelTransformer(), transformer],
+          });
+        }).toThrow(error);
+      });
+    });
+  });
+
+  describe('Valid usage', () => {
+    test.each([
+      {
+        name: 'accepts valid length validation configurations',
+        schema: /* GraphQL */ `
+          type Post @model {
+            id: ID!
+            title: String! @validate(type: minLength, value: "3")
+            content: String! @validate(type: maxLength, value: "10")
+          }
+        `,
+      },
+      {
+        name: 'accepts length validation on List field',
+        schema: /* GraphQL */ `
+          type Post @model {
+            id: ID!
+            tags: [String]! @validate(type: minLength, value: "20")
+            comments: [String]! @validate(type: maxLength, value: "30")
+          }
+        `,
+      },
+      {
+        name: 'accepts length values of "0"',
+        schema: /* GraphQL */ `
+          type Post @model {
+            id: ID!
+            title: String! @validate(type: minLength, value: "0")
+            content: String! @validate(type: maxLength, value: "0")
+          }
+        `,
+      },
+      {
+        name: 'accepts length values of extremely large numbers beyond 64-bit range',
+        schema: /* GraphQL */ `
+          type Post @model {
+            id: ID!
+            title: String! @validate(type: minLength, value: "999999999999999999999999999999")
+            content: String! @validate(type: maxLength, value: "999999999999999999999999999999")
+          }
+        `,
+      },
+    ])('$name', ({ schema }) => {
+      const out = testTransform({
+        schema,
+        transformers: [new ModelTransformer(), new ValidateTransformer()],
+      });
+      expect(out).toBeDefined();
+      expect(out.schema).toMatchSnapshot();
```

Comment:
Same as above. Consider checking specific types. Comparing full snapshot for too many cases would significantly increase the line counts in snapshot files.